src/components/subscriptionDialog.vue

```diff
@@ -0,0 +1,279 @@
+          data-cy="dismiss-modal"
+          text
+          @click="$store.dispatch('HANDLE_SUBSCRIPTION_DIALOG', false)"
+        >
+          Cancel
+        </v-btn>
+        <v-btn
+          color="primary"
+          text
+          id="confirm-btn"
+          data-cy="confirm-btn"
+          :disabled="!cardNumberIsValid"
+          @click="submitSubscription()"
+          style="display: flex;"
+        >
+          {{ loading ? 'Confirming Details' : 'Confirm Details' }}
+          <v-progress-circular
+            style="padding-left: 40px;"
+            indeterminate
+            v-if="loading"
+            :width="2.5"
+            :size="20"
+            color="primary"
+          />
+        </v-btn>
+      </v-card-actions>
+    </v-card>
+  </v-dialog>
+</template>
+
+<script>
+import {
+  HANDLE_SUBSCRIPTION_DIALOG, HANDLE_EDGE_SYNC_DIALOG, FETCH_USER_SUBSCRIPTION
+} from '@/store/action-types'
+import { mapActions, mapState } from 'vuex'
+const cardValidator = /^(?:4[0-9]{12}(?:[0-9]{3})?|[25][1-7][0-9]{14}|6(?:011|5[0-9][0-9])[0-9]{12}|3[47][0-9]{13}|3(?:0[0-5]|[68][0-9])[0-9]{11}|(?:2131|1800|35\d{3})\d{11})$/
+
+export default {
+  name: 'SubcriptionDialog',
+  data: () => ({
+    showInputs: false,
+    cardNumber: '',
+    expiryMonth: '',
+    expiryYear: '',
+    cvc: '',
+    fullName: '',
+    subscriptionError: null,
+
+    cardNumberIsValid: false,
+    emailAddress: '',
+    loading: false,
+    rules: {
+      required: value => !!value || 'Required.'
+    }
+  }),
+  methods: {
+    ...mapActions([HANDLE_SUBSCRIPTION_DIALOG, FETCH_USER_SUBSCRIPTION, HANDLE_EDGE_SYNC_DIALOG]),
+    async submitSubscription () {
+      this.loading = true
+
+      try {
+        const req = await fetch(`${process.env.VUE_APP_FUNCTIONS_ENDPOINT}/subscription`, {
+          method: 'POST',
+          body: JSON.stringify({
+            email: this.user.email,
+            number: this.cardNumber,
+            cvc: this.cvc,
+            exp_year: this.expiryYear,
+            exp_month: this.expiryMonth
+          }),
+          headers: {
+            'content-type': 'application/json'
+          }
+        })
+
+        const { userStripeId, userSubscriptionId } = await req.json()
+        this.saveStripeData(userStripeId, userSubscriptionId)
+      } catch (e) {
+        console.log(e, 'ERROR FROM STRIPE')
+        this.subscriptionError = e
+      } finally {
+        this.loading = false
+      }
+    },
+    async saveStripeData (userStripeId, userSubscriptionId) {
+      await fetch(`${process.env.VUE_APP_FUNCTIONS_ENDPOINT}/subscription-data`,
+        {
+          method: 'POST',
+          body: JSON.stringify({
+            userStripeId,
+            userSubscriptionId,
+            user_id: this.user.sub
+          }),
+          headers: {
+            'content-type': 'application/json'
+          }
+        }
+      )
+      this.loading = false
+      this.$store.dispatch(HANDLE_SUBSCRIPTION_DIALOG, false)
```

Comment:
@vickywane did you mean to `await` for these dispatch actions? Otherwise their order of execution is not guaranteed and this `saveStripeData` function will return immediately, before the actions complete their work.

Reply:
The asynchronous execution of some of the action do not matter to the application logic as some of them are background operations.  

Those whose order of execution matter I used the `await` keyword for. I have however made this correcton